docs/develop/create-fexpr.rst

```diff
@@ -0,0 +1,391 @@
+        Column rescol = evaluate1(awf.retrieve_column(i),
+                                  bwf.retrieve_column(i));
+        outputs.add_column(std::move(rescol), std::string(), gmode);
+      }
+      return outputs;
+    }
+
+The method ``evaluate1()`` will take a pair of two columns and produce
+the output column containing the result of ``gcd(a, b)`` calculation. We must
+take into account the stypes of both columns, and decide which stypes are
+acceptable for our function:
+
+.. code-block:: C++
+
+    Column evaluate1(Column&& a, Column&& b) const {
+      SType stype1 = a.stype();
+      SType stype2 = b.stype();
+      SType stype0 = common_stype(stype1, stype2);
+      switch (stype0) {
+        case SType::BOOL:
+        case SType::INT8:
+        case SType::INT16:
+        case SType::INT32: return make<int32_t>(std::move(a), std::move(b), SType::INT32);
+        case SType::INT64: return make<int64_t>(std::move(a), std::move(b), SType::INT64);
+        default:
+            throw TypeError() << "Invalid columns of types " << stype1 << " and "
+                << stype2 << " in " << repr();
+      }
+    }
+
+    template <typename T>
+    Column make(Column&& a, Column&& b, SType stype0) const {
+      a.cast_inplace(stype0);
+      b.cast_inplace(stype0);
+      return Column(new Column_Gcd<T>(std::move(a), std::move(b)));
+    }
+
+As you can see, the job of the ``FExpr_Gcd`` class is to produce a workframe
+containing one or more ``Column_Gcd`` virtual columns. This is where the actual
+calculation of GCD values will take place, and we shall declare this class too.
+It can be done either in a separate file in the `core/column/` folder, or
+inside the current file `expr/fexpr_gcd.cc`.
+
+.. code-block:: C++
+
+    #include "column/virtual.h"
+
+    template <typename T>
+    class Column_Gcd : public Virtual_ColumnImpl {
+      private:
+        Column acol_;
+        Column bcol_;
+
+      public:
+        Column_Gcd(Column&& a, Column&& b)
+          : Virtual_ColumnImpl(a.nrows(), a.stype()),
+            acol_(std::move(a)), bcol_(std::move(b))
+        {
+          xassert(acol_.nrows() == bcol_.nrows());
+          xassert(acol_.stype() == bcol_.stype());
+          xassert(compatible_type<T>(acol_.stype()));
+        }
+
+        ColumnImpl* clone() const override {
+          return new Column_Gcd(Column(acol_), Column(bcol_));
+        }
+
+        size_t n_children() const noexcept { return 2; }
+        const Column& child(size_t i) { return i==0? acol_ : bcol_; }
+
+        bool get_element(size_t i, T* out) {
+          T a, b;
+          bool avalid = acol_.get_element(i, &a);
+          bool bvalid = bcol_.get_element(i, &b);
+          if (avalid && bvalid) {
+            while (b) {
+              T tmp = b;
+              b = a % b;
+              a = tmp;
+            }
+            *out = a;
+            return true;
+          }
+          return false;
+        }
+    };
```

Comment:
Yeah, and those `acol_` / `bcol_` variables appear as hyperlinks. This is because GitHub doesn't know to stop syntax highlighting inside `.. code-block::` blocks.